### PR TITLE
Fix  aiomysql.sa in Python 3.7 and run tests for python 3.7 in CI

### PR DESCRIPTION
**Pull request by CeresCa**
## What do these changes do?
Fix async iterator protocol in aiomysql.utils._SAConnectionContextManager,
and run tests for python 3.7 in CI.

1. Add   `__aiter__` and  `__anext__` methods for `_SAConnectionContextManager`
2. Update .travis.yml for Python 3.7 

## Are there changes in behavior for the user?
No changes.

## Related issue number
[#410](https://github.com/aio-libs/aiomysql/issues/410)

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

**Comment by CeresCa** [comment]
@terrycain   I need someone to review this PR. 
It will be nice to see a new release built and published to PyPI after this PR merged.

File: .travis.yml [hunk 1]
```diff
@@ -13,26 +16,27 @@ services:
...
-              mariadb: 10.0
-        - python: 3.6
-          env: PYTHONASYNCIODEBUG=
-          addons:
-              mariadb: 10.1
-        - python: 3.6
-          env: PYTHONASYNCIODEBUG=
-          addons:
-              mysql: 5.7
+    include:
```
Comment (webknjaz) [inline .travis.yml]: Please revert the indentation changes. They are not related and make the review harder.

File: .travis.yml [hunk 2]
```diff
@@ -52,10 +56,10 @@ deploy:
...
         all_branches: true
-        python: 3.6
+        python: 3.7
 
 script:
     - make cov
 
 after_success:
-    - codecov
+    - codecov
```
Comment (webknjaz) [inline .travis.yml]: Unrelated.

File: aiomysql/utils.py [hunk 3]
```diff
@@ -86,7 +95,6 @@ class _TransactionContextManager(_ContextManager):
 
 
 class _PoolAcquireContextManager(_ContextManager):
-
```
Comment (webknjaz) [inline aiomysql/utils.py]: unrelated

File: tests/conftest.py [hunk 4]
```diff
@@ -332,7 +332,7 @@ def mysql_server(unused_port, docker, session_id, mysql_tag, request):
                         cursor.execute('FLUSH PRIVILEGES')
 
                 break
-            except Exception as err:
+            except Exception:
```
Comment (webknjaz) [inline tests/conftest.py]: unrelated

File: tests/test_issues.py [hunk 5]
```diff
@@ -76,7 +76,7 @@ def test_issue_8(self):
         yield from c.execute("drop table if exists test")
         yield from c.execute("""CREATE TABLE `test` (`station` int(10) NOT
             NULL DEFAULT '0', `dh`
-            datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
+            datetime NOT NULL DEFAULT '1000-01-01 00:00:00',
```
Comment (webknjaz) [inline tests/test_issues.py]: how is this related?

File: .travis.yml [hunk 6]
```diff
@@ -14,25 +16,26 @@ services:
 
 matrix:
    include:
-        - python: 3.6
+        - python: 3.7
```
Comment (webknjaz) [inline .travis.yml]: I don't think we should remove py3.6 jobs. For starters, it's better to add new ones.

**Comment by ethe** [comment]
Anything updates?

**Comment by webknjaz** [comment]
@CeresCa?

**Comment by terrycain** [comment]
Am busy with a new job but this projects been left for a while. Will have a look this coming week once ive fixed travis

**Comment by terrycain** [comment]
Remove the stuff from .travis matrix except from changing the main python version 3.6 to 3.7

**Comment by nickpro** [comment]
does this PR abandoned?

**Review by willbuckner — APPROVED** [review]

**Comment by Nothing4You** [comment]
thank you for the PR.

python 3.7 support has been implemented quite a while ago, therefore this is no longer relevant.

**Comment by Nothing4You** [comment]
for the SA fix, this should be covered in #494, which also includes a test case.
